tests/test_peak_properties.py

```diff
@@ -0,0 +1,98 @@
+import strax
+import numpy as np
+from hypothesis import given, strategies, example, settings
+import tempfile
+
+
+@settings(max_examples=100, deadline=None)
+@given(
+    # number of peaks
+    strategies.integers(min_value=0, max_value=20),
+    # length of the data field in the peaks
+    strategies.integers(min_value=2, max_value=20),
+)
+def test_index_of_fraction(peak_length, data_length):
+    """
+    Test strax.index_of_fraction
+    """
+    # Setup the peaks
+    dtype = [(('Start time since unix epoch [ns]', 'time'), np.int64),
+             (('Exclusive end time since unix epoch [ns]', 'endtime'), np.int64),
+             (('length of p', 'length'), np.int16),
+             (('area of p', 'area'), np.float64),
+             (('data of p', 'data'), (np.float64, data_length))]
+    peaks = np.zeros(peak_length, dtype=dtype)
+    peaks['time'] = np.arange(peak_length)
+    peaks['endtime'] = peaks['time'] + 1
+
+    # Fill the peaks with random length data
+    for p in peaks:
+        length = np.random.randint(0, data_length)

```

Comment:
Can do but as I don't see the how this would affect thinks I consider it unlikely that we ever have to replicate on this specific point. Therefore, I will keep it as it is.

Reply:
Turns out that this was a good comment though. See https://github.com/AxFoundation/strax/pull/363. Someday will rewrite this test in a similar way!